syntax="proto3";
package User;


//登录消息
message ReLogin
{

   string name=1;
   string pswd=2;

}
//登录返回消息
message RqLogin
{

     int32 code=1;

}
message ReRegister
{
   string name=1;
   string pswd=2;
   string phone=3;
   string email=4;
}
message RqRegister
{
    int32 code=1;
}


//角色信息
message Rqrole
{
    string roleName=1;
    int32 exp=2;
    int32 money=3;
    int32 level=4;
}
message addrole
{
    string roleName=1;
}

//返回状态码，
//0成功
//-1失败
//-2数据库错误
//-3未登录
//4匹配完成
//5匹配中
//6匹配取消
//7匹配确认完成
message rqcode
{
    int32 code=1;
}

message room_id
{
    int32 id=1;
}
//开始匹配
message matching
{
    int32 peopleNum=1;

}

message to_room
{
    int32 room_id=1;
}

message cilck_hero
{
 int32 room_id=1;
 int32 heroid=2;
}


//确定选择英雄
message select_hero
{
    int32 room_id=1;
    int32 heroid=2;
}
//房间角色信息
message role_hore
{
    string roleName=1; //名字
    int32 cilckhero=2; //点击的英雄
    int32 ranks=3; //队伍
    int32 index=4; //位置
    int32 heroid=5;  //选择英雄
    bool toroom=6;  //是否进入房间
}
//返回房间信息
message roominfo
{
repeated role_hore role_list=1;

}
message hero
{
    int32  heroid=1;
    string heroname=2;
    string describe=3;
}
message hero_all
{
    repeated hero hero_list=1;
}


//发送给所有用户的消息,只有一个消息
message re_message_all
{
    string message=1;
}

//发送队友或者，房间的消息，需要一个房间号，来找到这个房间
message re_message_room
{   int32 roomid=1;
    string message=2;
}

//返回的消息给其他用户
message send_message
{

    string name=1; //发送改消息的名字
    string message=2;//消息内容
}